packages/ifpack2/src/Ifpack2_Details_Filu_def.hpp

```diff
@@ -101,7 +101,8 @@ initLocalPrec()
 {
   auto nRows = this->mat_->getLocalNumRows();
   auto& p = this->params_;
-  localPrec_ = Teuchos::rcp(new LocalFILU(this->localRowPtrs_, this->localColInds_, this->localValues_, nRows, p.sptrsv_algo,
+  bool skipSortMatrix = !p.use_metis;
```

Comment:
I'm realizing now that this will sometimes skip sorting when it's still required. If the input matrix ``mat_`` is not a Tpetra::CrsMatrix and metis is off, skipSortMatrix will be true but we don't really know that the matrix is sorted.

```suggestion
  bool skipSortMatrix = !rcp_dynamic_cast<const TCrsMatrix>(mat_).is_null() && rcp_dynamic_cast<const TCrsMatrix>(mat_)->getCrsGraph()->isSorted() && !p.use_metis;
```

Reply:
You could also put the dynamic cast on its own line.